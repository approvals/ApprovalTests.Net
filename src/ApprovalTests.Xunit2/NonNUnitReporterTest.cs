using ApprovalTests.Reporters.TestFrameworks;
using ApprovalUtilities.Utilities;
using Xunit;

namespace ApprovalTests.Xunit2.Reporters
{
    public class NonNUnitReporterTest
    {
        [Fact]
        public void TestNunitIsNotWorkingFromXUnit()
        {
            Approvals.SetCaller();
            Assert.False(NUnitReporter.INSTANCE.IsWorkingInThisEnvironment("default.txt"));
        }

        [Fact]
        public void TestXunitIsWorking()
        {
            Approvals.SetCaller();
            Assert.True(XUnit2Reporter.INSTANCE.IsWorkingInThisEnvironment("default.txt"));
        }

        [Fact]
        public void TestXunitReporterIsWorking()
        {
            var e = ExceptionUtilities.GetException(() => XUnit2Reporter.INSTANCE.AssertEqual("Hello", "Hello2"));
            Approvals.Verify(e.Message);
        }
    }
}