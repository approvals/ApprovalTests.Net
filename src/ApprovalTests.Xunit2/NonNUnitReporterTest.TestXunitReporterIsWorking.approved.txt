Assert.Equal() Failure
               ? (pos 5)
Expected: Hello
Actual:   Hello2
               ? (pos 5)